giles-eco-tardis/src/main/java/edu/asu/diging/tardis/core/service/impl/InnogenScriptRunner.java

```diff
@@ -0,0 +1,62 @@
+package edu.asu.diging.tardis.core.service.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.nio.file.Path;
+
+import edu.asu.diging.gilesecosystem.septemberutil.properties.MessageType;
+import edu.asu.diging.gilesecosystem.septemberutil.service.ISystemMessageHandler;
+import edu.asu.diging.gilesecosystem.util.properties.IPropertiesManager;
+import edu.asu.diging.tardis.config.Properties;
+import edu.asu.diging.tardis.core.exception.InnogenScriptRunnerException;
+import edu.asu.diging.tardis.core.service.IInnogenScriptRunner;
+
+@Service
+public class InnogenScriptRunner implements IInnogenScriptRunner{
+    
+    @Autowired
+    protected ISystemMessageHandler messageHandler;
+    
+    @Autowired
+    protected IPropertiesManager propertiesManager;
+    
+    @Override
+    public void runInnogenScript(String imagePath, int pareNr) throws InnogenScriptRunnerException {
```

Comment:
should that be `pageNr`?

Reply:
actually, do you really need the page nr? it's only to create a unique path, right? why not generate ids instead?

Reply:
having pageNr is necessary because unlike the other services we don't get the whole pdf here but we get the images. Thus we need the pageNr in the request to set ,
https://github.com/diging/giles-eco-tardis/pull/1/files#diff-f69f21a3515614d1bf005e45bd2562e0cbca7c0d84a3e403fedc52b903211f03R130

Also I thought having the pageNr will make it a cleaner implementation as the images extracted from a particular page will be in that folder. And we access it using the pageNr in the request
https://github.com/diging/giles-eco-tardis/pull/1/files#diff-f69f21a3515614d1bf005e45bd2562e0cbca7c0d84a3e403fedc52b903211f03R123

Reply:
Hm, not sure I understand. Carolus doesn't get the whole file either and I believe the resulting files are associated via the file id. For the internal workings of the component it shouldn't matter (and would be clearer if there is no dependency on the page number).